enhancements/olm/runtime-bundle-configuration.md

```diff
@@ -0,0 +1,858 @@
+account for it when generating the final set of manifests. The given configuration must conform to the configuration schema
+provided by the bundle. If the bundle cannot be configured, or the configuration does not strictly comply with the
+schema, an error is raised that prevents installation/upgrade/update from proceeding until the user can address the
+issue, e.g.:
+- User provides configuration but the bundle does not support configuration
+- User does not provide required configuration fields
+- User provides configuration keys that are not existent
+- User provides configuration but bundle does not provide a configuration schema - note that 
+while for registry+v1 bundles this is deemed a good level of restriction to ship the configuration feature with initially, for future bundle formats this restriction may be loosed (eg OLMv1 could allow users to provide configuration for "Helm bundles" without any configuration schema). Users will likely be required to express their intent to configure such bundles explicity and have OLM install anyway, possibly via a ClusterExtension API field (eg `dangerouslyAcceptBundlesWithoutConfigSchema`). Such considerations however are for future considerations. 
+
+### registry+v1 Bundle Configuration Generation, Validation and Handling
+
+registry+v1 bundles do not carry a configuration schema and expecting the schema to exist in the FBC would mean only 
+those catalogs that have been updated to carry this information would benefit from the changes proposed by this enhancement. 
+Therefore, the decision was made to have OLM derive the configuration by observing the bundle configuration. 
+It is possible that this logic moves to the catalog layer. But, since other bundle sources are envisioned (e.g. direct bundle install), 
+for a first take, the following changes will go in the runtime:
+
+1. Update the registry+v1 manifest provisioning layer to generate a configuration schema for the bundle
+2. Update the registry+v1 manifest provisioning layer to validate user provided configuration against the generated schema returning any failures (and thereby aborting the operation)
+3. Update the registry+v1 bundle rendering layer to accept the `watchNamespace` configuration for those bundles that support it and render the manifests appropriately
+
+### Workflow Description
+
+The standard OLMv1 installation and upgrade workflows stay largely the same. Where it will differ is:
+- users will need to consult the [OpenShift documentation](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html-single/extensions/index#olmv1-deploying-a-ce-in-a-specific-namespace_managing-ce) to understand how packages/bundles can be configured. The OpenShift documentation will include a capability matrix (derived from the table in the "registry+v1 Configuration Schema" section below) that maps bundle install mode support to configuration requirements, specifically:
```

Comment:
Just a note that the currently linked documentation does not explain the new configuration interface and instead relies on an annotation-based API for configuring the watch namespace.

We should ensure this documentation is updated to reflect the current state of the feature usage and how an end-user determines what is and is not valid to put in the configuration.